### PR TITLE
add cancancan gem and update affected files

### PR DESCRIPTION
**Pull request by kian1213**
Hi sir, please review my pull request.My initial implentation to cancancan gem. Thanks po

### CODE REVIEW

File: app/controllers/admin_controller.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class AdminController < ApplicationController
 
   private
   def ensure_user_rights
-    unless user_signed_in? && current_user.admin?
+    unless user_signed_in? && current_user.is?(:admin)
```
Comment (kian1213) [inline app/controllers/admin_controller.rb]: I get error when I remove ()

**Comment by kian1213** [comment]
Ms. Grace is not yet replying.

File: app/models/user.rb [hunk 2]
```diff
@@ -4,6 +4,8 @@ class User < ActiveRecord::Base
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
 
+  enum user_type: {admin: "admin", supervisor: "supervisor", trainee: "trainee"}
```
Comment (dieunb) [inline app/models/user.rb]: pls, read enum again. Value in model is mapping with value in database as integer.
```
enum user_type: {admin: 0, supervisor: 1, trainee: 2}
```

File: app/views/admin/users/_form.html.erb [hunk 3]
```diff
@@ -33,17 +33,11 @@
   <div class="form-group">
     <%= f.label :user_type, class: "col-md-2 control-label" %>
     <div class="col-md-10">
-      <%= f.select :user_type, ["supervisor", "trainee"], {},
+      <%= f.select :user_type, @user_type.each_with_index{|v, k| [v, k]}, {},
```
Comment (dieunb) [inline app/views/admin/users/_form.html.erb]: try `User.user_types`

File: app/views/admin/users/_form.html.erb [hunk 4]
```diff
@@ -33,17 +33,11 @@
   <div class="form-group">
     <%= f.label :user_type, class: "col-md-2 control-label" %>
     <div class="col-md-10">
-      <%= f.select :user_type, ["supervisor", "trainee"], {},
+      <%= f.select :user_type, ["admin", "supervisor", "trainee"], {},
```
Comment (kian1213) [inline app/views/admin/users/_form.html.erb]: I just revert it the old way,, it still save 0, 1, 2, in user_type db

File: app/models/ability.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize user
+    if user.admin?
+      can :manage, :all
+    elsif user.is? :supervisor
```
Comment (dieunb) [inline app/models/ability.rb]: try `user.supervisor?`

**Pull request merged**